silk/tests/openthread/ot_test_address_cache_table.py

```diff
@@ -0,0 +1,435 @@
+# limitations under the License.
+"""Test address cache table gets properly updated in case a node moves to a different parent
+   in the same thread network.
+"""
+
+import random
+import time
+import unittest
+
+from silk.config import wpan_constants as wpan
+from silk.node.wpan_node import WpanCredentials
+from silk.utils import process_cleanup
+from silk.tools import wpan_table_parser
+from silk.tools.wpan_util import verify, verify_within, VerifyError
+from silk.unit_tests.test_utils import random_string
+import silk.hw.hw_resource as hwr
+import silk.node.fifteen_four_dev_board as ffdb
+import silk.tests.testcase as testcase
+
+hwr.global_instance()
+
+PREFIX = "fd00:1234::"
+POLL_INTERVAL = 400
+ROUTER_TABLE_WAIT_TIME = 30
+INVALID_ROUTER_ID = 63
+CHILD_SUPERVISION_CHECK_TIMEOUT = 2
+PARENT_SUPERVISION_INTERVAL = 1
+REATTACH_WAIT_TIME = CHILD_SUPERVISION_CHECK_TIMEOUT
+WAIT_TIME = 180
+
+
+class TestAddressCacheTable(testcase.TestCase):
+    # Build network topology
+    #
+    #     r1 ---- r2 ---- r3
+    #     |       |       |
+    #     |       |       |
+    #     fed1    sed2   fed3
+    #
+    # fed1 and fed3 are FED children, sed2 is an SED which is first attached to r2 and
+    # then forced to switch to r3.
+
+    r2_rloc, r3_rloc, fed3_rloc = None, None, None
+    r1_address, fed1_address, sed2_address, fed3_address = None, None, None, None
+
+    @classmethod
+    def hardware_select(cls: 'TestAddressCacheTable'):
+        cls.r1 = ffdb.ThreadDevBoard()
+        cls.r2 = ffdb.ThreadDevBoard()
+        cls.r3 = ffdb.ThreadDevBoard()
+        cls.fed1 = ffdb.ThreadDevBoard()
+        cls.sed2 = ffdb.ThreadDevBoard()
+        cls.fed3 = ffdb.ThreadDevBoard()
+
+        cls.all_nodes = [cls.r1, cls.r2, cls.r3, cls.fed1, cls.sed2, cls.fed3]
+
+    @classmethod
+    @testcase.setup_class_decorator
+    def setUpClass(cls):
+        # Check and clean up wpantund process if any left over
+        process_cleanup.ps_cleanup()
+
+        cls.hardware_select()
+
+        for device in cls.all_nodes:
+            device.set_logger(cls.logger)
+            cls.add_test_device(device)
+            device.set_up()
+
+        cls.network_data = WpanCredentials(network_name="SILK-{0:04X}".format(random.randint(0, 0xffff)),
+                                           psk="00112233445566778899aabbccdd{0:04x}".format(random.randint(0, 0xffff)),
+                                           channel=random.randint(11, 25),
+                                           fabric_id="{0:06x}dead".format(random.randint(0, 0xffffff)))
+
+        cls.thread_sniffer_init(cls.network_data.channel)
+
+    @classmethod
+    @testcase.teardown_class_decorator
+    def tearDownClass(cls: 'TestAddressCacheTable'):
+        for device in cls.device_list:
+            device.tear_down()
+
+    @testcase.setup_decorator
+    def setUp(self):
+        pass
+
+    @testcase.teardown_decorator
+    def tearDown(self):
+        pass
```

Comment:
Are these needed if they don't do anything?

Reply:
Will check and update on this.

Reply:
yes these are required as without these test cases do not start. i.e. after setup class directly teardown class is executed.